celery/beat.py

```diff
@@ -257,7 +258,9 @@ def tick(self, event_t=event_t, min=min, heappop=heapq.heappop,
         adjust = self.adjust
         max_interval = self.max_interval
 
-        if self._heap is None:
+        if (self._heap is None or
+                not self.schedules_equal(self.old_schedulers, self.schedule)):
```

Comment:
These `self.old_schedulers` and `self.schedule` are of type `dict`, so the `==` or `!=` is only a shallow comparison.